### PR TITLE
HRINT-2673 - FTSQuerySearchBasics java

### PR DESCRIPTION
**Pull request by Drorlev**

### CODE REVIEW

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/FTSQuerySearchBasics.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        //region Demo
+        List<Employee> employeesWithMatchingTerms;
+
+        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
+            //region Step_1
+            employeesWithMatchingTerms = session.query(Employee.class)
+            //endregion
+                //region Step_2
+                .search("Notes", term1)
```
Comment (ml054) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/FTSQuerySearchBasics.java:24]: please use single search here: 
```
.search("Notes", term1 + " " + term2)
```

**Comment by ml054** [comment]
@Danielle9897 could you review markdown part?

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                "  .search('Notes', 'searchTerm')",
+                "",
+                "* In this example, we search for _term1 OR term2_ in the document field _Notes_.",
+                "",
+                "* The default operator between _search_ and _search_ is OR.<br>",
+                "  See the related operations demo link for setting this to an AND operation.",
+                "",
+                //Somthing better could be written in 72 - 73?
+                "* Since only a single Search() method is used,<br> then documents containing 'term1' will come first in the results before those containing 'term2'.<br>",
+                "  Use Boosting to influence results order when using multiple Search() methods."
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json:68]: Fix in 2 places above:

Search() => search()

**Review by Danielle9897 — CHANGES_REQUESTED** [review]

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/FTSQuerySearchBasics.java (hunk 1 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/FTSQuerySearchBasics.java:24]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json (hunk 2 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json:68]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/FTSQuerySearchBasics.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        //region Demo
+        List<Employee> employeesWithMatchingTerms;
+
+        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
+            //region Step_1
+            employeesWithMatchingTerms = session.query(Employee.class)
+            //endregion
+                //region Step_2
+                .search("Notes", term1 + " " + term2)
+                //.search("Notes", term2)
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/FTSQuerySearchBasics.java:25]: remove the old line with the comment ?

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            "Slug": "step-2",
+            "DescriptionLines": [
+                "* Use `search()` to search for the specified term in the specified document field.",
+                "",
+                "* The most basic usage is to search for a single term, i.e.:<br>",
+                "  .search('Notes', 'searchTerm')",
+                "",
+                "* In this example, we search for _term1 OR term2_ in the document field _Notes_.",
+                "",
+                "* The default operator between _term1 and term2_ is OR.<br>",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json:63]: _term1 and term2_ => 'term1' and 'term2'

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/FTSQuerySearchBasics.java (hunk 3 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/FTSQuerySearchBasics.java]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json (hunk 4 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json:63]: Done

**Comment by Drorlev** [comment]
I added the FTSQuerySearchOperators

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                    "Type": "demo",
+                    "Url": "/java/queries/query-overview"
+                }
+            ]
+        },
+        {
+            "Title": "Define search criteria",
+            "Slug": "step-2",
+            "DescriptionLines": [
+                "* Use `search()` to search for the specified term in the specified document field.",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json:56]: term => terms

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+                "",
+                "* The most basic usage is to search for a single term, i.e.:<br>",
+                "  .search('Notes', 'searchTerm')",
+                "",
+                "* In this example, we search for 'term1' and 'term2' in the document field _Notes_.",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json:61]: ` 'term1' and 'term2'` =>  `term1 OR term2`

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json (hunk 4 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json:63]: this shoud be:
`'term1' and 'term2'`

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json (hunk 6 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json:61]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json (hunk 5 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json:56]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json (hunk 4 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json:63]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        {
+            "Title": "Define search criteria",
+            "Slug": "step-2",
+            "DescriptionLines": [
+                "* Use `search()` to search for the specified terms in the specified document field.",
+                "",
+                "* The most basic usage is to search for a single term, i.e.:<br>",
+                "  .search('Notes', 'searchTerm')",
+                "",
+                "* In this example, we search for term1 OR term2 in the document field _Notes_.",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchBasics/metadata.json:61]: need to make it with an underscore - just like in the C# demo

`term1 OR term2` => `_term1 OR term2_`

**Review by Danielle9897 — APPROVED** [review]

**Review by ml054 — APPROVED** [review]

**Pull request merged**